File: Sources/Sentry/SentryEnvelopeItemHeader.m
#import "SentryEnvelopeItemHeader.h"

@implementation SentryEnvelopeItemHeader

- (instancetype)initWithType:(NSString *)type length:(NSUInteger)length
{
    if (self = [super init]) {
        _type = type;
        _length = length;
    }
    return self;
}

- (instancetype)initWithType:(NSString *)type
                      length:(NSUInteger)length
                 contentType:(NSString *)contentType
{
    if (self = [self initWithType:type length:length]) {
        _contentType = contentType;
    }
    return self;
}

- (instancetype)initWithType:(NSString *)type
                      length:(NSUInteger)length
                   filenname:(NSString *)filename
                 contentType:(NSString *)contentType
{
    if (self = [self initWithType:type length:length contentType:contentType]) {
        _filename = filename;
    }
    return self;
}

- (instancetype)initWithType:(NSString *)type
                      length:(NSUInteger)length
                 contentType:(NSString *)contentType
                   itemCount:(NSNumber *)itemCount
{
    if (self = [self initWithType:type length:length contentType:contentType]) {
        _itemCount = itemCount;
    }
    return self;
}

- (NSDictionary *)serialize
{

    NSMutableDictionary *target = [[NSMutableDictionary alloc] init];
    if (self.type) {
        [target setValue:self.type forKey:@"type"];
    }

    if (self.filename) {
        [target setValue:self.filename forKey:@"filename"];
    }

    if (self.contentType) {
        [target setValue:self.contentType forKey:@"content_type"];
    }

    if (self.platform) {
        [target setValue:self.contentType forKey:@"platform"];
    }

    if (self.itemCount) {
        [target setValue:self.itemCount forKey:@"item_count"];
    }

    [target setValue:[NSNumber numberWithUnsignedInteger:self.length] forKey:@"length"];

    return target;
}

@end

File: Sources/Swift/Tools/SentryEnvelopeItemHeader.swift
// swiftlint:disable missing_docs
import Foundation

@_spi(Private) @objc public class SentryEnvelopeItemHeader: NSObject, SentrySerializable {

    /**
     * The type of the envelope item.
     */
    @objc public let type: String
    @objc public let length: UInt
    @objc public let filename: String?
    @objc public let contentType: String?
    @objc public let itemCount: NSNumber?

    /**
     * Some envelopes need to report the platform name for enhanced rate limiting functionality in
     * relay.
     */
    @objc public var platform: String?

    @objc public init(type: String, length: UInt) {
        self.type = type
        self.length = length
        self.filename = nil
        self.contentType = nil
        self.itemCount = nil
        super.init()
    }

    @objc public init(type: String, length: UInt, contentType: String?) {
        self.type = type
        self.length = length
        self.filename = nil
        self.contentType = contentType
        self.itemCount = nil
        super.init()
    }

    @objc public init(type: String, length: UInt, filename: String?, contentType: String?) {
        self.type = type
        self.length = length
        self.filename = filename
        self.contentType = contentType
        self.itemCount = nil
        super.init()
    }

    @objc public init(type: String, length: UInt, contentType: String?, itemCount: NSNumber?) {
        self.type = type
        self.length = length
        self.filename = nil
        self.contentType = contentType
        self.itemCount = itemCount
        super.init()
    }

    @objc public func serialize() -> [String: Any] {
        var target = [String: Any]()

        target["type"] = type
        target["filename"] = filename
        target["content_type"] = contentType
        target["platform"] = platform
        target["item_count"] = itemCount
        target["length"] = NSNumber(value: length)

        return target
    }
}
// swiftlint:enable missing_docs
